### PR TITLE
KAZUI-262: re-indent whole project with tabs, standardize on size 4 tabs

### PR DESCRIPTION
**Pull request by danielfinke**
split this into 2 commits - https://github.com/2600hz/kazoo-ui/commit/872f53bb5b669611134ccab23ffd9b822cc1ddf3 actually adds the rulesets, the other commit is just updating all the indentation.
Using tabs mostly just to be consistent with Monster-UI - the .editorconfig and .eslintrc.yml are from Monster-UI and adapted/trimmed down. More rules could be added back in later but this is just focussed on indentation.
@nunovieira you will be interested in this

### CODE REVIEW

**Comment by danielfinke** [comment]
@nunovieira I disagree that the inclusion of a package.json makes something a Node.js app - if the app itself isn't being run with `node` it rather simply indicates that there is some Node.js utility being used for building, testing, linting, or otherwise in the project.
I do however agree that it's less than ideal to include it but I'll lay out my reasoning for doing so:

1. Some developers may use editors that do not support EditorConfig out of the box (e.g. stock vim), and this would inevitably lead to the project becoming a mix of styles again
2. The provision of eslint rules gives a quick way to test for conformity after edits have been made, and eslint won't let you run plugins without them being in a local `node_modules` folder
3. We could potentially share our Buildkite and/or CircleCI automated tests that are built on top of eslint - easier than writing custom scripts for just testing indentation when there is already a perfectly good linter - extra protection from the project becoming a mix of styles again

Depending on how you're shipping/packaging the app, you could remove or restrict access to node_modules/package.json/package-lock.json (in the case of a git deployment) or simply exclude from RPMs/etc. (we use https://github.com/jordansissel/fpm for packaging, very simple exclusions config)

**Comment by nunovieira** [comment]
@danielfinke ok then. Looks like the config files need some tuning to make this good to merge.

File: js/winkstart.js [hunk 1]
```diff
@@ -1,143 +1,143 @@
...
+		loading = {},
+		locked_topics = {},
+		slice = [].slice;
+
+	winkstart.get_version = function(callback) {
+		$.ajax({
+			url: 'VERSION',
+			cache: amplify.cache,
+			success: function(template) {
             	template = template.replace(/[\n\s]/g,'');
```
Comment (nunovieira) [inline js/winkstart.js]: Some places like this one and comments where not properly indented.

File: .eslintrc.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  - promise
+env:
+  browser: true
+  jquery: true
+
+globals:
+  winkstart: true
+  form2object: true
+  _t: true
+  Sentry: true
```
[the remark this answers is not in the document]
Comment (danielfinke) [reply .eslintrc.yml]: Good catch, forgot to pull this one out (we use Sentry for analytics)

File: .eslintrc.yml [hunk 3]
```diff
@@ -0,0 +1,17 @@
+# Original source from OpenTelecom:ci-linting, mira-t on github
+plugins:
+  - standard
+  - promise
```
[the remark this answers is not in the document]
Comment (danielfinke) [reply .eslintrc.yml]: Just an omission from cleanup when moving the MUI one here. Will remove it

File: .editorconfig [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+indent_style = tab
+
+# Indentation Size
+indent_size = tab
+
+# Width of a single tabstop character
+tab_width = 4
+
+# Line ending file format (lf|crlf|cr)
+# end_of_line = crlf
```
[the remark this answers is not in the document]
Comment (danielfinke) [reply .editorconfig]: I also added the `linebreak-style` rule to .eslintrc.yml.

**Comment by danielfinke** [comment]
All reviews addressed

**Comment by azefiel** [comment]
Looks good to me.

Just noticed this but not sure why you've added `package-lock.json` as a `.gitignore` entry though?

**Comment by danielfinke** [comment]
@azefiel agree that it should be in the repo - added it

**Pull request merged**